File: go/raindrops/raindrops.go
// Package raindrops exposes a function that has to do with raindrops and stuff.
package raindrops

import "fmt"

var dropMap = map[int]string{
	3: "Pling",
	5: "Plang",
	7: "Plong",
}

// Convert is kinda useless, but I need to write these docs to keep the bots happy.
func Convert(input int) string {

	var output string

	for i := 3; i <= 7; i += 2 {
		if input%i == 0 {
			output += dropMap[i]
		}
	}

	if output == "" {
		return fmt.Sprintf("%d", input)
	}

	return output
}

File: swift/raindrops/Sources/Raindrops.swift
//Solution goes in Sources

struct Raindrops {

    let value: Int

    init(_ value: Int) {
        self.value = value
    }

    var sounds: String {

        var output: String = ""

        if value % 3 == 0 {
            output.append("Pling")
        }

        if value % 5 == 0 {
            output.append("Plang")
        }

        if value % 7 == 0 {
            output.append("Plong")
        }

        if output == "" {
            output.append("\(value)")
        }

        return output
    }
}
